Code_Exercises/Matrix_Transpose/solution.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ SYCL Academy (c)
+
+ SYCL Academy is licensed under a Creative Commons
+ Attribution-ShareAlike 4.0 International License.
+
+ You should have received a copy of the license along with this
+ work.  If not, see <http://creativecommons.org/licenses/by-sa/4.0/>.
+
+ The tiling should work as follows:
+
+ The groupOffset will need to be inverted, as well as the localId.
+
+ In:                                      Out:
+ +----------------------------+           +----------------------------+
+ |             group  ^       |           |       ^                    |
+ |           Offset[1]|       |           |       |                    |
+ |                    V       |           |       |                    |
+ |<------------------>+-------+           |       |                    |
+ |   groupOffset[0]   |1     2|           |       |groupOffset[0]      |
+ |                    | tile  |           |       |                    |
+ |                    |3     4| ------->  |       |                    |
+ |                    +-------+           |       V                    |
+ |                            |           |       +-------+            |
+ |                            |           |       |1     3|            |
+ |                            |           |       | tile  |            |
+ |                            |           |       |2     4|            |
+ +----------------------------+           +-------+-------+------------+
+                                           <----->
+                                           groupOffset[1]
+
+Within a tile, each work item is assigned to a single value:
+
+InTile:               OutTile:
++------------+        +------------+
+|   local  ^ |        |  local^    |
+|   Id[1]  | |        |  Id[0]|    |
+|          V |------->|       |    |
+|<-------->* |        |       V    |
+|localId[0]  |        |<----->*    |
++------------+        +------------+
+                      localId[1]
+
+*/
+
+#include <iostream>
+#include <vector>
+
+#if __has_include(<sycl/sycl.hpp>)
```

Comment:
We can probably just simplify this to sycl/sycl.hpp now

Reply:
Good catch thanks